perfkitbenchmarker/benchmark_spec.py

```diff
@@ -344,8 +344,15 @@ def PrepareVm(self, vm):
     logging.info('Waiting for boot completion.')
     vm.AllowRemoteAccessPorts()
     vm.WaitForBootCompletion()
-    vm.AddMetadata(benchmark=self.name, perfkit_uuid=self.uuid,
-                   benchmark_uid=self.uid)
+    vm_metadata = {'benchmark': self.name,
+                   'perfkit_uuid': self.uuid,
+                   'benchmark_uid': self.uid}
+    if FLAGS.vm_metadata:
```

Comment:
Since the default is an empty list (and not none), I don't think you need this guard